packages/camera_driver/include/camera_driver_node/raspberry_pi_64.py

```diff
@@ -30,79 +23,67 @@ async def worker(self):
 
         Captures a frame from the /dev/video0 image sink and publishes it.
         """
-        if self._device is None or not self._device.isOpened():
+        if self._camera is None:
             self.logerr("Device was found closed")
             return
         # init queues
         await self.dtps_init_queues()
         # get first frame
-        retval, image = self._device.read() if self._device else (False, None)
-
+        numpy_array = self._camera.capture_array() if self._camera else None
+        if self.configuration.rotation:
+            numpy_array = numpy.rot90(numpy_array)
+        success, jpeg_encoded_numpy_array = cv2.imencode('.jpeg', numpy_array)
         # keep reading
         while not self.is_shutdown:
-            if not retval:
+            if success is None:
                 self.logerr("Could not read image from camera")
                 await asyncio.sleep(1)
-                continue
-            if image is not None:
-                # without HW acceleration, the image is returned as RGB, encode on CPU
-                jpeg: bytes = cast(np.ndarray, image).tobytes()
-
+            else:
+                jpeg: bytes = jpeg_encoded_numpy_array.tobytes()
                 # publish
                 await self.publish(jpeg)
             # return control to the event loop
             await asyncio.sleep(0.001)
             # grab next frame
-            retval, image = self._device.read() if self._device else (False, None)
+            numpy_array = self._camera.capture_array() if self._camera else None
+            if self.configuration.rotation:
+                numpy_array = numpy.rot90(numpy_array)
+            success, jpeg_encoded_numpy_array = cv2.imencode('.jpeg', numpy_array)
         self.loginfo("Camera worker stopped.")
 
     def setup(self):
         # setup camera
-        cam_props = {
-            "video_bitrate": 25000000,
+        self._camera = Picamera2()
+        main = {
+            "size": (self.configuration.res_w, self.configuration.res_h),
```

Comment:
This is a breaking change with the configuration format, where we specified width and height after performing the rotation of the image.